src/test/java/uk/gov/pay/api/it/fixtures/PaymentResultBuilder.java

```diff
@@ -174,10 +162,11 @@ protected TestPaymentFailureState(String status, String message, String code) {
     protected String returnUrl;
     protected String paymentProvider;
     protected String createdDate;
-    protected List<ImmutableMap<?, ?>> links = new ArrayList<>();
+    protected List<Map<?, ?>> links = new ArrayList<>();
```

Comment:
It's because we should generally code to interfaces and not worry about the implementation (`ImmutableMap`, `HashMap` etc) here.

Reply:
Agreed, unless the implementation or more constrained interface is important, e.g. that for some reason the immutability was required in this instance; I don't see why it would be here other than immutability being favoured when possible so happy with the change.